### PR TITLE
Switched coordinates overlay display as (y,x) in RTL languages

### PR DESCRIPTION
**Pull request by aaka3207**
1. Changed the `coordinatesProvider` code used in the _TooltipOverlay_ component to swap the direction of the coordinates displayed when isRTL is true.
2. Added `isRTL` property from redux store to all components that use `coordinatesProvider`
3. Fixed issue with tooltip overlay clipping outside of canvas on RTL languages.

**Before**
before

**After (no background due to my limited s3 access)**
after

Part of [FND-1488
](https://codedotorg.atlassian.net/secure/RapidBoard.jspa?rapidView=28&projectKey=FND&modal=detail&selectedIssue=FND-1488)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by aaka3207** [comment]
tagging @dju90

**Comment by dju90** [comment]
From the screenshot, it looks like there's still an offset where the origin is not being calculated from the lower left corner. You can see the x-coordinate dashed line is still not going to the edge of the display area. Besides the right-to-left coordinate text which you already fixed, it should have the same display behavior as the left-to-right coordinate system/display, shown here in this gif (note the origin in the lower left corner)
ltr-coords

**Comment by dju90** [comment]
Also, I'm looking at the coordinate text tag and it looks a little wonky:
`y, 92 :x: 163`
It should instead be
`92 :y ,163 :x`
(see https://www.quora.com/In-right-to-left-languages-are-chart-axes-drawn-differently for reference)

**Comment by aaka3207** [comment]
> From the screenshot, it looks like there's still an offset where the origin is not being calculated from the lower left corner. You can see the x-coordinate dashed line is still not going to the edge of the display area. Besides the right-to-left coordinate text which you already fixed, it should have the same display behavior as the left-to-right coordinate system/display, shown here in this gif (note the origin in the lower left corner)
> ltr-coords

It seems my change didn't stick, the reason was simply because the `visualizationOverlay `needed a `left: auto` style attribute instead of `left: 0`. I'm gonna put the attribute inline and see if that works.

**Comment by aaka3207** [comment]
> Also, I'm looking at the coordinate text tag and it looks a little wonky:
> `y, 92 :x: 163`
> It should instead be
> `92 :y ,163 :x`
> (see https://www.quora.com/In-right-to-left-languages-are-chart-axes-drawn-differently for reference)

I'm actively trying to figure this out, as I've actually never seen this behavior before. For some reason when the coordinate is to the left of the colon and the axis is to the right, combined with two coordinates they push against each other. You can see that it it's actual value is here

image

**Comment by aaka3207** [comment]
@dju90  here is an updated screencap:

before_canvas

File: apps/src/templates/TooltipOverlay.jsx [hunk 1]
```diff
@@ -160,9 +160,22 @@ export function textProvider(label) {
...
+      return [
+        '\u202A',
+        Math.round(y),
+        ' :y',
+        ', ',
+        Math.round(props.mouseX),
+        ' :',
+        'x',
+        '\u202C'
+      ].join(''); // We have to use unicode to explicitly set the SVG text to LTR when the documen is using RTL
```
Comment (Hamms) [inline apps/src/templates/TooltipOverlay.jsx]: I'd much prefer to see this implemented using interpolation

```suggestion
      // We have to use unicode to explicitly set the SVG text to LTR when the document is using RTL
      return `\u202A${Math.round(y)} :y, ${Math.round(props.mouseX)} :x\u202C`;
```

**Review by Hamms — APPROVED** [review]
one implementation nit, otherwise LGTM!

File: apps/src/templates/TooltipOverlay.jsx (hunk 1 above)
Comment (aaka3207) [reply apps/src/templates/TooltipOverlay.jsx]: @Hamms when I was testing the unicode it doesn't change when using interpolation with backticks. I could do concatenation with + but not sure how much of an improvement that is

Comment (Hamms) [reply apps/src/templates/TooltipOverlay.jsx]: hmmm, that shouldn't be the case. Quick manual testing indicates that those two approaches produce completely identical strings:

```js
one = ['\u202A', 1, ' :y', ', ', 2, ' :', 'x', '\u202C'].join('');
two = `\u202A${1} :y, ${2} :x\u202C`;

console.log(one === two);
```

What behavior difference do you see when using interpolation vs concatenation?

**Pull request merged**